### DISCUSSION TITLE
Open modal automatically after page component loaded

### DISCUSSION THREAD

**Discussion by howdu** [wire-elements/modal#263] [Q&A] [answered]
In a standard livewire page component how can I automatically open a modal?

Tried this but nothing hapens:
```
public function render()
    {
        $return = parent::render();

        if (session('upload-facebook')) {
            $this->emit('openModal', 'videos.upload.facebook-upload', [session('upload-facebook')]);
        }

        return $return;
    }
```

Reason being that I have a table of videos with a button that opens a modal.

Before opening the modal it calls another action which checks the user has permision to post to facebook. If not it will redirect to facebook. 

The problem I have is that when redreting back from facebook how can I trigger the modal to open automatically without the user having to click the modal button again.

**Comment by PhiloNL** [comment]
I think it will work if you add it to the blade file of your component instead:

```blade
@if((session('upload-facebook')))
    <script>Livewire.emit('openModal', 'videos.upload.facebook-upload', [session('upload-facebook')])</script>
@endif
```

**Comment by Random72IsTaken** [comment, marked as answer]
```Blade
@if ($browseNewsId = session('autoBrowseNewsId'))
    @push('scripts')
        <script>
            document.addEventListener('alpine:initialized', () => {
                Livewire.emit("openModal", "show-news", {!! json_encode(["newsId" => $browseNewsId]) !!});
            });
        </script>
    @endpush
@endif
```
